File: reference/src/core/compare.ts
/**
 * Shared comparison utilities used across policy modules.
 */

/**
 * Compare a numeric value against a threshold using the given operator.
 */
export function compareMetric(actual: number, operator: string, threshold: number): boolean {
  switch (operator) {
    case '>=':
      return actual >= threshold;
    case '<=':
      return actual <= threshold;
    case '==':
      return actual === threshold;
    case '!=':
      return actual !== threshold;
    case '>':
      return actual > threshold;
    case '<':
      return actual < threshold;
    default:
      return false;
  }
}

/** Severity ordering for tool-rule evaluation. */
const SEVERITY_ORDER: Record<string, number> = {
  low: 1,
  medium: 2,
  high: 3,
  critical: 4,
};

/**
 * Return true if `actual` severity exceeds `max` severity.
 */
export function exceedsSeverity(
  actual: 'low' | 'medium' | 'high' | 'critical',
  max: 'low' | 'medium' | 'high' | 'critical',
): boolean {
  return SEVERITY_ORDER[actual] > SEVERITY_ORDER[max];
}

File: sdk-python/src/ai_sdlc/core/compare.py
"""Shared comparison utilities used across policy modules."""

from __future__ import annotations

from typing import Literal

Severity = Literal["low", "medium", "high", "critical"]

_SEVERITY_ORDER: dict[str, int] = {
    "low": 1,
    "medium": 2,
    "high": 3,
    "critical": 4,
}


def compare_metric(actual: float, operator: str, threshold: float) -> bool:
    """Compare a numeric value against a threshold using the given operator."""
    match operator:
        case ">=":
            return actual >= threshold
        case "<=":
            return actual <= threshold
        case "==":
            return actual == threshold
        case "!=":
            return actual != threshold
        case ">":
            return actual > threshold
        case "<":
            return actual < threshold
        case _:
            return False


def exceeds_severity(actual: Severity, max_severity: Severity) -> bool:
    """Return True if ``actual`` severity exceeds ``max_severity``."""
    return _SEVERITY_ORDER[actual] > _SEVERITY_ORDER[max_severity]
